vsintegration/src/FSharp.Editor/Completion/FileSystemCompletion.fs

```diff
@@ -89,38 +93,61 @@ module internal FileSystemCompletion =
         else
             None
 
-type internal LoadDirectiveCompletionProvider() =
+    let private includeDirectiveCleanRegex = Regex("""#I\s+(@?"*(?<literal>[^"]*)"?)""", RegexOptions.Compiled ||| RegexOptions.ExplicitCapture)
+
+    let getIncludeDirectives (document: Document, position: int) =
+        async {
+            let! ct = Async.CancellationToken
+            let! text = document.GetTextAsync(ct)
+            let lines = text.Lines
+            let caretLine = text.Lines.GetLinePosition(position).Line
+            return
+                lines
+                |> Seq.filter (fun x -> x.LineNumber <= caretLine)
+                |> Seq.choose (fun line ->
+                    let lineStr = line.ToString().Trim()
+                    // optimization: fail fast if the line does not start with "(optional spaces) #I"
+                    if not (lineStr.StartsWith "#I") then None
```

Comment:
It works inside comment blocks. why not? :) Disabling it requires querying lexer.